packages/react-slider/src/components/Slider/useSliderState.tsx

```diff
@@ -344,15 +358,33 @@ export const useSliderState = (state: SliderState) => {
     const marksValue = markValues;
     const marksChildren: JSX.Element[] = [];
     for (let i = 0; i < marksPercent.length; i++) {
+      const marksItem = typeof marks === 'boolean' ? null : marks![i];
+
       marksChildren.push(
         <div className={markContainerClassName} key={`markItemContainer-${i}`}>
-          <div
-            className={mergeClasses(
-              markClassName,
-              (marksValue[i] === 0 && firstMarkClassName) || (marksValue[i] === 100 && lastMarkClassName) || '',
-            )}
-            key={`mark-${i}`}
-          />
+          {marksItem && typeof marksItem === 'object' && marksItem.mark ? (
+            marksItem.mark
+          ) : (
+            <div
+              className={mergeClasses(
+                markClassName,
+                (marksValue[i] === 0 && firstMarkClassName) || (marksValue[i] === 100 && lastMarkClassName) || '',
+              )}
+              key={`mark-${i}`}
+            />
+          )}
+          {marksItem !== (undefined || null) &&
+            typeof marksItem === 'object' &&
+            marksItem.label &&
+            (typeof marksItem.label === 'string' ? (
+              <Label className={markLabelClassName} key={`markLabel-${i}`} disabled={disabled}>
+                {marksItem.label}
+              </Label>
```

Comment:
That is a fair question. The nice thing about using the `Label` component is that it would be able to handle the disabled state and apply styling which would require extra code otherwise.

Reply:
Coming back to this I think that we should remove the `Label`. It simplifies the logic in the state file and drops the bundle size 2-3 kB.